Gems/AtomTressFX/Code/Rendering/HairFeatureProcessor.cpp

```diff
@@ -316,6 +316,62 @@ namespace AZ
                 return pass ? true : false;
             }
 
+            bool HairFeatureProcessor::CreateHairParentPass(RPI::RenderPipeline* renderPipeline)
+            {
+                if (HasHairParentPass(renderPipeline))
+                {
+                    CreatePerPassResources();
+                    return false;
+                }
+                const char* passRequestAssetFilePath = "Passes/AtomTressFX_PassRequest.azasset";
+                m_hairPassRequestAsset = AZ::RPI::AssetUtils::LoadAssetByProductPath<AZ::RPI::AnyAsset>(
+                    passRequestAssetFilePath, AZ::RPI::AssetUtils::TraceLevel::Warning);
+                const AZ::RPI::PassRequest* passRequest = nullptr;
+                if (m_hairPassRequestAsset->IsReady())
+                {
+                    passRequest = m_hairPassRequestAsset->GetDataAs<AZ::RPI::PassRequest>();
+                }
+                if (!passRequest)
+                {
+                    AZ_Warning("AtomTressFx", false, "Failed to load PassRequest from %s", passRequestAssetFilePath);
+                    return false;
+                }
+
+                m_usePPLLRenderTechnique = passRequest->m_templateName == AZ::Name("HairParentPassTemplate");
+
+                 // Create the pass
+                RPI::Ptr<RPI::Pass> hairParentPass  = RPI::PassSystemInterface::Get()->CreatePassFromRequest(passRequest);
+
+                // add the pass to render pipeline
+                RPI::Ptr<RPI::ParentPass> rootPass = renderPipeline->GetRootPass();
+
+                // Find opaque pass
+                auto passFilter = RPI::PassFilter::CreateWithTemplateName(Name("OpaqueParentTemplate"), renderPipeline);
+                RPI::Ptr<RPI::Pass> opaquePass = nullptr;
+                RPI::PassSystemInterface::Get()->ForEachPass(passFilter, [&opaquePass](RPI::Pass* pass) -> RPI::PassFilterExecutionFlow
+                    {
+                        opaquePass = pass;
+                        return RPI::PassFilterExecutionFlow::StopVisitingPasses;
+                    });
+
+                if (!opaquePass)
+                {
+                    return false;
+                }
+                // insert the pass 
+                auto parentPass = opaquePass->GetParent();
+                auto passIndex = parentPass->FindChildPassIndex(Name("OpaquePass"));
+                    
+                bool success = parentPass->InsertChild(hairParentPass, passIndex.GetIndex()+1);
```

Comment:
! You're using passIndex without checking if it's valid, which it may not be if parentPass can't find a child named "OpaquePass". If it's invalid, value will be uint max, and adding one will overflow to 0, so you'll be adding the hair parent pass to the start of the renderpipeline.
Simple check to see if passIndex is valid will fix this.